pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -64,6 +65,24 @@ func (strategy) CheckGracefulDelete(obj runtime.Object, options *kapi.DeleteOpti
 	return false
 }
 
+type statusStrategy struct {
+	strategy
+}
+
+var StatusStrategy = statusStrategy{Strategy}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status.
+func (statusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDc := obj.(*api.DeploymentConfig)
+	oldDc := old.(*api.DeploymentConfig)
+	newDc.Spec = oldDc.Spec
+}
+
+// ValidateUpdate is the default update validation for an end user updating status.
+func (statusStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	return validation.ValidateDeploymentConfigUpdate(obj.(*api.DeploymentConfig), old.(*api.DeploymentConfig))
```

Comment:
I think I want a separate ValidateDeploymentConfigStatusUpdate method... I don't think we want invalid things in the spec to make status updates fail